lib/cloud_controller/runner.rb

```diff
@@ -153,6 +153,12 @@ def setup_telemetry_logging
 
     def setup_db
       db_logger = Steno.logger('cc.db')
+
+      if @config.get(:webserver) == 'puma'
+        max_db_connections_per_process = @config.get(:puma, :max_db_connections_per_process)
+        @config.set(:db, @config.get(:db).merge(max_connections: max_db_connections_per_process)) unless max_db_connections_per_process.nil?
```

Comment:
One small thing: I find this config.get + config.set + config.get confusing... What about:
```
db_config = @config.get(:db)

if @config.get(:webserver) == 'puma'
  max_db_connections_per_process = @config.get(:puma, :max_db_connections_per_process)
  db_config.merge!(max_connections: max_db_connections_per_process)) unless max_db_connections_per_process.nil?
end

DB.load_models(db_config, db_logger)
```